File: mongobee_migration_java/src/main/java/com/github/nkolytschew/mongodbmigration/dao/jpa/repository/UserRepository.java
package com.github.nkolytschew.mongodbmigration.dao.jpa.repository;


import com.github.nkolytschew.mongodbmigration.dao.jpa.document.UserDocument;

import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.data.rest.core.annotation.RepositoryRestResource;

/**
 * simple Repository.
 * Use {@link RepositoryRestResource} to avoid creating custom controller.
 */
@RepositoryRestResource(path = "users")
public interface UserRepository extends MongoRepository<UserDocument, String> {
}

File: mongobee_migration/src/main/kotlin/com/github/nkolytschew/mongobee_migration/dao/jpa/repository/UserRepository.kt
package com.github.nkolytschew.mongobee_migration.dao.jpa.repository

import com.github.nkolytschew.mongobee_migration.dao.jpa.document.UserDocument
import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.data.rest.core.annotation.RepositoryRestResource

/**
 * simple Repository.
 * Use [RepositoryRestResource] to avoid creating custom controller.
 */
@RepositoryRestResource(path = "users")
interface UserRepository : MongoRepository<UserDocument, String>